src/properties/struct_property.rs

```diff
@@ -184,40 +187,36 @@ impl PropertyTrait for StructProperty {
                 StructPropertyValue::CustomStruct(type_name, _) => type_name,
             })?;
             cursor.write_all(&self.guid.0)?;
-            cursor.write_all(&[0u8; 1])?;
+            cursor.write_u8(0)?;
             write_begin = cursor.stream_position()?;
         }
 
         match &self.value {
             StructPropertyValue::Vector(vector) => {
-                FloatProperty::new(f32::from(vector.x)).write(cursor, false)?;
-                FloatProperty::new(f32::from(vector.y)).write(cursor, false)?;
-                FloatProperty::new(f32::from(vector.z)).write(cursor, false)?;
+                cursor.write_f32::<LittleEndian>(f32::from(vector.x))?;
```

Comment:
I think the logic of doing this may be reversed. We are de-serializing a `StructProperty`, not a `FloatProperty`. The current code works because calling `FloatProperty::write(_, false)` coincidentally only makes one `write_f32` call; but changes to `FloatProperty::write` could cause issues for `StructProperty`.

The actual objective here is to improve compiler optimization - it does not always optimize the `::read(_, false)` call to a naked `write_f32` call, because the call needs to be inline before if(include_header) can be considered a dead branch, and the compiler does not reliably inline the call.

Reply:
UE5s float being double precision could mess this up so I don't think changing it is a great idea, I think a tradeoff of extensibility vs perf. should be done

Reply:
Okay, I will remove this from the branch for now, as we can re-visit this code in the context of LWC types when UE5 support is added if we need to.